languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -155,25 +189,165 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
 ;;  Member Declarations
 ;; =====================
 
+[
+  (class_declaration)
+  (enum_declaration)
+  (field_declaration)
+  (interface_declaration)
+  (method_declaration)
+  (constructor_declaration)
+  (annotation_type_declaration)
+  (constant_declaration)
+  (record_declaration)
+] @decl
+{
+  node @decl.defs
+  node @decl.lexical_scope
+}
+
+(annotation_type_declaration
+  name: (identifier) @name) @annotation {
+
+  node def
+  attr (def) node_definition = @name
+  edge def -> @annotation.lexical_scope
+  edge @annotation.defs -> def
+}
+
+(constructor_declaration body: (constructor_body) @body) @this {
+  edge @body.lexical_scope -> @this.lexical_scope
+}
+
+(constructor_body) @this {
+  node @this.lexical_scope
+}
+
+(constructor_body . (_) @first) @this {
+  edge @first.before_scope -> @this.lexical_scope
+}
+
+(constructor_body (_) @a . (_) @b) {
+  edge @b.before_scope -> @a.after_scope
+}
+
+(explicit_constructor_invocation) @this {
+  node @this.before_scope
+  node @this.after_scope
+}
+
+(enum_declaration name: (_) @name) @this {
+  node def
+  attr (def) node_definition = @name
+  edge @this.defs -> def
+
+  node @this.constants
+  attr (@this.constants) pop_symbol = "."
+  edge def -> @this.constants
+
+  ; allow A.X field accesses to resolve constants
+  node def__typeof
+  attr (def__typeof) pop_symbol = ":"
+  edge def -> def__typeof
+  edge def__typeof -> @this.constants
+}
```

Comment:
This is a tiny bit funny. We use `:` push/pop marker nodes to signal that something—like a field or method—should be looked up on a type. We synthesize such for the static members—well, the constants, anyway—of `enum`s so that `A.X` (where `A` is an `enum` and `X` is one of its constants), which syntactically is parsed as a field access (as though `A` were an object and `X` one of its fields), resolves without every field access resulting in redundant paths.

This isn't the tidiest way to organize the graph, and it feels a bit… gross, I guess, abusing `:` for statics when it really ought to be about typing. But then maybe that's what statics are; if you think about metaclasses and all that jazz from Ruby/Smalltalk/Objective-C, it starts to sound familiar.

At any rate, we will probably use a similar approach for static methods whenever we implement them correctly (they're out of scope for this PR).